io/azure.go

```diff
@@ -0,0 +1,120 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/container"
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/azureblob"
+)
+
+// Constants for Azure configuration options
+const (
+	AdlsSasTokenPrefix         = "adls.sas-token."
+	AdlsConnectionStringPrefix = "adls.connection-string."
+	AdlsSharedKeyAccountName   = "adls.auth.shared-key.account.name"
+	AdlsSharedKeyAccountKey    = "adls.auth.shared-key.account.key"
+	AdlsEndpoint               = "adls.endpoint"
+	AdlsProtocol               = "adls.protocol"
+
+	// Not in use yet
+	// AdlsReadBlockSize          = "adls.read.block-size-bytes"
+	// AdlsWriteBlockSize         = "adls.write.block-size-bytes"
+)
+
+// Construct a Azure bucket from a URL
+func createAzureBucket(ctx context.Context, parsed *url.URL, props map[string]string) (*blob.Bucket, error) {
+	adlsSasTokens := propertiesWithPrefix(props, AdlsSasTokenPrefix)
+	adlsConnectionStrings := propertiesWithPrefix(props, AdlsConnectionStringPrefix)
+
+	// Construct the client
+	accountName := props[AdlsSharedKeyAccountName]
+	endpoint := props[AdlsEndpoint]
+	protocol := props[AdlsProtocol]
+
+	var client *container.Client
+
+	if accountName != "" {
```

Comment:
let's make this more readable by inverting this. 

```go
if accountName == "" {
    return nil, errors.New("account name is required for azure bucket")
}

// rest of code
```

Just makes it easier to read

Reply:
done.